app/src/main/java/com/kamron/pogoiv/ShareHandlerActivity.java

```diff
@@ -0,0 +1,37 @@
+package com.kamron.pogoiv;
+
+import android.app.Activity;
+import android.content.Intent;
+import android.graphics.Bitmap;
+import android.net.Uri;
+import android.provider.MediaStore;
+import android.support.v4.content.LocalBroadcastManager;
+import android.os.Bundle;
+
+import java.io.IOException;
+
+public class ShareHandlerActivity extends Activity {
+
+    @Override
+    protected void onCreate(Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_share_handler);
+        finish();
+
+        Intent intent = getIntent();
+        String action = intent.getAction();
+        String type = intent.getType();
+
+        if (Intent.ACTION_SEND.equals(action) && type != null && type.startsWith("image/")) {
+            Uri imageUri = (Uri) intent.getParcelableExtra(Intent.EXTRA_STREAM);
+            try {
+                Bitmap bitmap = MediaStore.Images.Media.getBitmap(this.getContentResolver(), imageUri);
+                Intent newintent = new Intent("process-bitmap");
+                newintent.putExtra("bitmap", bitmap);
+                LocalBroadcastManager.getInstance(ShareHandlerActivity.this).sendBroadcast(newintent);
```

Comment:
> Any idea if the actual bitmap associated with the Uri will still be kept around if I finish handling the this intent from Google Now? 

No idea I'm afraid; I'm not familiar with how it works at all. 

When I ran across that exception the way I fixed it was to write the bitmap to disk and pass a URI.  Thinking about it, I was dealing with larger bitmaps, say 4,000*3,000 pixels. This could be fine.